### PR TITLE
Improve click arguments and options in download.py

### PR DESCRIPTION
**Pull request by danielfernau**
- use click.Path() to check if given download destination path exists and is writable and to resolve to an absolute path (+ remove os/path import)
- show defaults for click options (defaults are now visible when using `--help`)
- add interactive prompt for CloudKey IP address if not provided as command line argument
- add interactive prompt for username if not provided as command line argument
- add interactive, hidden prompt for password if not provided as command line argument
- parse `--start` and `--end` arguments as datetime using click.DateTime()
- update option help texts
- remove outdated code
- reformat file

### CODE REVIEW

File: protect_archiver/cli/download.py [hunk 1]
```diff
@@ -1,111 +1,152 @@
...
 from ..client import ProtectClient, ProtectError
 
 
 @cli.command("download", help="Download footage from a local UniFi Protect system")
-@click.argument("dest")
-@click.option("--address", default="unifi", help="CloudKey IP address or hostname")
-@click.option("--port", default=7443, help="UniFi Protect service port")
+@click.argument(
+    "dest",
+    type=click.Path(exists=True, writable=True, resolve_path=True)
```
Comment (dcramer) [inline protect_archiver/cli/download.py]: 👌

File: protect_archiver/cli/download.py [hunk 2]
```diff
@@ -1,111 +1,152 @@
...
+)
+@click.option(
```
Comment (dcramer) [inline protect_archiver/cli/download.py]: did you not want to keep the default?

I'm assuming for almost everyone its `unifi`

File: protect_archiver/cli/download.py [hunk 3]
```diff
@@ -1,111 +1,152 @@
...
-    download_wait,
-    download_timeout,
-    use_subfolders,
-    touch_files,
-    skip_existing_files,
-    ignore_failed_downloads,
-    start,
-    end,
-    create_snapshot,
+        dest,
```
Comment (dcramer) [inline protect_archiver/cli/download.py]: indentation here probably came from black fwiw

`black .` will run it on the codebase - we could also setup pre-commit if wanted to try to consistently do it

Comment (danielfernau) [reply protect_archiver/cli/download.py]: Using `black` now, too. Thanks for the hint. Format should be as expected now 😄

File: protect_archiver/cli/download.py (hunk 2 above)
Comment (danielfernau) [reply protect_archiver/cli/download.py]: `unifi` didn't resolve to an IP within my network when I tried it - turns out my DNS server wants a full local domain name for it to work. Re-added it as the default and removed the interactive prompt.

**Pull request merged**